pkg/api/errors/errors.go

```diff
@@ -297,6 +297,17 @@ func NewGenericServerResponse(code int, verb, kind, name, serverMessage string,
 	case len(kind) > 0:
 		message = fmt.Sprintf("%s (%s %s)", message, strings.ToLower(verb), kind)
 	}
+	var causes []api.StatusCause
+	if isUnexpectedResponse {
+		causes = []api.StatusCause{
+			{
+				Type:    api.CauseTypeUnexpectedServerResponse,
+				Message: serverMessage,
+			},
+		}
+	} else {
+		causes = nil
```

Comment:
Couldnt think of a cause to add. The current ones are specific to field value validation: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/types.go#L1496
Feel free to suggest, if you have something in mind.